### PR TITLE
Handle `echo` BIDS tag in `fetch_subject_and_session`

### PR DESCRIPTION
**Pull request by NathanMolinier**
## Description

For some subjects, an `echo` BIDS tag can be present in the filename. See example in sct_testing_large:
```bash
sub-zurichMPM05_ses-01_echo-1_T1w.nii.gz
sub-zurichMPM05_ses-01_echo-1_T1w.json
sub-zurichMPM05_ses-01_echo-2_T1w.nii.gz
sub-zurichMPM05_ses-01_echo-2_T1w.json
sub-zurichMPM05_ses-01_echo-3_T1w.nii.gz
sub-zurichMPM05_ses-01_echo-3_T1w.json
```

This PR aims to fetch this information for potential future use.

### CODE REVIEW

File: manual_correction.py [hunk 1]
```diff
@@ -756,7 +756,7 @@ def main():
                     time.sleep(0.1)
                     print("")
                     # build file names
-                    subject, ses, filename, contrast = utils.fetch_subject_and_session(file)
+                    subject, ses, filename, contrast, echo = utils.fetch_subject_and_session(file)
                     # Construct absolute path to the input file
                     # For example: '/Users/user/dataset/data_processed/sub-001/anat/sub-001_T2w.nii.gz'
                     fname = os.path.join(path_img, subject, ses, contrast, filename)
```
Comment (valosekj) [inline manual_correction.py:762]: Shouldn't the `echo` tag be included here?

Comment (valosekj) [reply manual_correction.py:762]: Okay, probably not, because it is already part of `filename`.

Comment (NathanMolinier) [reply manual_correction.py:762]: It's strange because it seems to be there for me

**Comment by valosekj** [comment]
Good point! Thank you for pointing this out! But I am wondering if treating the `echo` BIDS tag is really necessary because the tag is always included in the `filename` variable (see https://github.com/spinalcordtoolbox/manual-correction/pull/63#discussion_r1320031018). Also, the `echo` is not used in folder structure (unlike sub and ses, for example, `sub-001/ses-01`). So, there is actually no need to fetch the `echo` tag because it is never used. It is something similar to the `acq` tag; the script is already compatible with it without the need to fetch it.

**Comment by valosekj** [comment]
> I agree, this is also why I don't like the suffix `acq` ahah. Personally, this information was useful to differentiate files when we have the same `subject name`, `contrast` and `session`. I thought that this information could be relevant in this function because we are gathering information contained within the filename. However, if you feel like this function is only useful to retrieve the BIDS folders, I can understand.

We fetch `subject`, `ses`, and `contrast` because they are used to construct the directory structure:

https://github.com/spinalcordtoolbox/manual-correction/blob/9e97e664ebe4281b23aa8e2c6291e0a000d277a4/manual_correction.py#L759-L762

To differentiate between files, we also fetch `filename` containing the `acq` or `echo` tag.

**Comment by NathanMolinier** [comment]
Then maybe this PR is not so necessary 😄

**Comment by valosekj** [comment]
Discussed in person with @NathanMolinier. Closing without merge since fetching the `echo` tag is not needed right now.

**Pull request closed without merging**